packages/traceability/issuer-irec-api-wrapper/src/IRECAPIClient.ts

```diff
@@ -482,13 +482,16 @@ export class IRECAPIClient extends EventEmitter {
     }
 
     public async redeem(redemption: Redemption): Promise<RedeemTransactionResult> {
+        const r =
+            redemption instanceof Redemption ? redemption : plainToClass(Redemption, redemption);
```

Comment:
I wonder - why it's required? It messed the code.

Theoretically class will be JSON.stringified by axios which will remove stuff like functions, and will use all strigifiable fields by default.

You can customize this behavior by using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description (`.toJSON`) - which I have mixed opinion about.

Also, if interface is not giant, it may be good idea to just manually  select all the fields you want to send.

Reply:
Honestly, I do not understand you

If you ask why we use `plainToClass`, we need to apply data transforms for things like dates.